File: src/main/kotlin/org/kite9/diagram/visualization/orthogonalization/flow/FaceVertex.java
package org.kite9.diagram.visualization.orthogonalization.flow;

import org.kite9.diagram.common.elements.edge.Edge;
import org.kite9.diagram.common.elements.edge.PlanarizationEdge;
import org.kite9.diagram.common.elements.vertex.Vertex;
import org.kite9.diagram.visualization.planarization.Face;

public class FaceVertex {

	Face face;
	Vertex vertex;
	Edge prior;
	Edge after;

	@Override
	public boolean equals(Object arg0) {
		if (arg0 instanceof FaceVertex) {
			FaceVertex ff = (FaceVertex) arg0;
			return (ff.face == face) && (ff.vertex == vertex) && (ff.prior == prior) && (ff.after == after);
		}
		return false;
	}

	@Override
	public int hashCode() {
		return face.hashCode() + vertex.hashCode() + prior.hashCode() + after.hashCode();
	}

	public String toString() {
		return face.getID() + "-" + vertex.getID() + "-" + prior + getUnderlying(prior) + "-" + after + getUnderlying(after);
	}

	private String getUnderlying(Edge e) {
		return ((PlanarizationEdge)e).getDiagramElements().keySet().toString();
	}
}

File: src/main/kotlin/org/kite9/diagram/visualization/orthogonalization/flow/FaceVertex.kt
package org.kite9.diagram.visualization.orthogonalization.flow

import org.kite9.diagram.common.elements.edge.PlanarizationEdge
import org.kite9.diagram.common.elements.vertex.Vertex
import org.kite9.diagram.visualization.planarization.Face

data class FaceVertex (val face: Face, val vertex: Vertex, val prior: PlanarizationEdge,val after: PlanarizationEdge) {

    override fun toString(): String {
        return face!!.getID() + "-" + vertex!!.getID() + "-" + prior + getUnderlying(prior) + "-" + after + getUnderlying(
            after
        )
    }

    private fun getUnderlying(e: PlanarizationEdge): String {
        return e.getDiagramElements().keys.toString()
    }
}